src/main/groovy/org/mockito/release/notes/vcs/RevisionDateProvider.java

```diff
@@ -12,16 +13,23 @@
 class RevisionDateProvider {
 
     private final ProcessRunner runner;
+    /**
+     * pattern for validating revision date in ISO format
+     * Example of valid date: 2017-01-29 08:14:09 -0800
+     */
+    private final Pattern REVISION_DATE_PATTERN = Pattern.compile(
+            "\\s?\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\s[-+]\\d{4}\\s?");
 
     RevisionDateProvider(ProcessRunner runner) {
         this.runner = runner;
     }
 
     public Date getDate(String rev) {
         String gitOutput = runner.run("git", "log", "--pretty=%ad", "--date=iso", rev, "-n", "1");
-        //TODO process runner needs to fail on error here.
-        // Otherwise the 'date' variable holds some error message and not really any date.
-        //Example output returned by running git command: 2017-01-29 08:14:09 -0800
+        if(!REVISION_DATE_PATTERN.matcher(gitOutput).matches()){
+           throw new IllegalArgumentException("Can't get a proper date for revision number " + rev + ". Are you sure this revision exists?");
```

Comment:
Minor, it might be good to call out 'tag' explicitly here to make it easy to grasp by users: "Are you sure this revision exists?" -> "Are you sure this revision or tag exists?"